setup.py

```diff
@@ -1008,7 +1098,7 @@ def build_extensions(self):
             raise DistutilsError(
                 'None of TensorFlow, PyTorch, or MXNet plugins were built. See errors above.')
 
-require_list = ['cloudpickle', 'psutil', 'six']
+require_list = ['cmake', 'cloudpickle', 'psutil', 'six']
```

Comment:
Can we depend on cmake only if the system does not have one?  Do we know of minimal version required?

Reply:
I think maybe python setup will only install packages when it's missing? According to gloo, the minimal version is 2.8.12.

Reply:
cmake is usually installed by `apt install` or other system package manager.  `pip install cmake` is a clever trick, but it can overwrite system-installed cmake.

Reply:
So do you mean that we remove the cmake from python setup requirement and specify it as an out-of-band requirement that user needs to install themselves?

Reply:
I think what Alex is suggesting here is that we test to see if cmake is already installed (for example, try running `cmake` as we do elsewhere in `setup.py`), and if it's not, only then do we add cmake to the list of required packages to install.

Reply:
Yep, that's exactly right.